### PR TITLE
feat:add GetListTTL

### PR DESCRIPTION
**Pull request by wangxuanni**
get remaining time to live of a key that has a timeout ,see https://github.com/nutsdb/nutsdb/issues/304

### CODE REVIEW

File: tx_list.go [hunk 1]
```diff
@@ -338,3 +338,10 @@ func (tx *Tx) CheckExpire(bucket string, key []byte) bool {
 	}
 	return false
 }
+
+func (tx *Tx) GetListTTL(bucket string, key []byte) (uint32, error) {
+	if err := tx.checkTxIsClosed(); err != nil {
+		return 0, err
+	}
+	return tx.db.ListIdx[bucket].GetListTTL(string(key))
```
Comment (elliotchenzichang) [inline tx_list.go:346]: Hi, Seems it shows this line is not covered by the unit test. Would you like to add a unit test to cover it? Thanks in advance.

File: ds/list/list.go [hunk 2]
```diff
@@ -439,3 +439,19 @@ func (l *List) IsEmpty(key string) (bool, error) {
 	}
 	return true, nil
 }
+
+func (l *List) GetListTTL(key string) (uint32, error) {
+	if l.IsExpire(key) {
+		return 0, ErrListNotFound
+	}
+	ttl := l.TTL[key]
+	timestamp := l.TimeStamp[key]
```
Comment (elliotchenzichang) [inline ds/list/list.go:448]: Seems the TTL implementation consists of two times in the codebase. It is not very reasonable to me, maybe I lack some context here. So would you or @xujiajun like to kindly have a discussion talk about the implementation of the TTL feature with me when you have time? Thanks in advance.

**Review by elliotchenzichang — COMMENTED** [review]
Left two questions. Please have a look when you have time. Thanks.

**Comment by elliotchenzichang** [comment]
Thanks for your contributions.

File: tx_list.go (hunk 1 above)
Comment (wangxuanni) [reply tx_list.go:346]: Already commited based on your suggestion, thanks！

**Pull request merged**